jujugui/static/gui/src/app/components/search-results/item/item.js

```diff
@@ -195,6 +201,57 @@ YUI.add('search-results-item', function(Y) {
     },
 
     /**
+      Closes the search results.
+
+      @method _close
+    */
+    _close: function() {
+      this.props.changeState({
+        search: null,
+        profile: null
+      });
+    },
+
+    /**
+      Deploy the entity.
+
+      @param entity {Object} The entity to deploy.
+      @param evt {Object} The click event.
+    */
+    _handleDeploy: function(entity, evt) {
+      if (entity.type === 'charm') {
```

Comment:
By passing all of this logic into here it's really duplicating a lot of existing complexity. Have you looked at just passing the `deployTarget` method into here? 

https://github.com/juju/juju-gui/blob/develop/jujugui/static/gui/src/app/app.js#L2233

This method is called by DD as well to extract all the complexity away from adding it to the canvas. It requires the charmstore as the second parameter which will make it difficult to pre-bind, so you may want to swap the order of the arguments first.